modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/react-fragments/src/select-project/index.js

```diff
@@ -39,72 +39,72 @@ const handleClickHome = () => {
 export default function () {
 	const [items, setItems] = useState([]);
 	const [value, setValue] = useState('');
-	const [
-		selectedExternalReferenceCode,
-		setSelectedExternalReferenceCode,
-	] = useState('');
+	const selectedExternalReferenceCode = useMemo(() => {
+		const liferaySearchParams = new URLSearchParams(window.location.search);
+
+		return liferaySearchParams.get('kor_id');
+	}, []);
 
 	const filteredItems =
 		items?.filter((item) =>
 			item?.label?.toLowerCase().includes(value?.toLowerCase())
 		) || [];
 
-	const firstItems = items.filter((_, index) => index <= 19);
-
-	useEffect(() => {
-		const liferaySearchParams = new URLSearchParams(window.location.search);
+	if (!filteredItems.length) {
+		filteredItems.push({
+			className: 'px-3 my-3 text-neutral-5',
+			disabled: true,
+			label: 'No projects match that name.',
+		});
+	}
 
-		setSelectedExternalReferenceCode(liferaySearchParams.get('kor_id'));
-	}, []);
+	const firstItems = items.filter((_, index) => index <= 19);
```

Comment:
firstItems is not a good name, because is relative.

In the future, let's find a better name

Reply:
I think, firstItems should be, something like, 1,2,3 items, not 19.

It could be: firstPageItems